### PR TITLE
Use fix-sized arrays in AddPool and UpdatePool tns

### PR DESCRIPTION
**Pull request by Dominator008**
Avoid dynamic types to allow consistent abi encoding.

### CODE REVIEW

File: contracts/libraries/Transitions.sol [hunk 1]
```diff
@@ -517,11 +517,11 @@ library Transitions {
             bytes32 stateRoot,
             uint32 poolId,
             uint32 strategyId,
-            uint32[] memory rewardAssetIds,
-            uint256[] memory rewardPerEpoch,
+            uint32[5] memory rewardAssetIds,
+            uint256[5] memory rewardPerEpoch,
```
Comment (xiaozhou) [inline contracts/libraries/Transitions.sol:521]: MAX_REWARD_ASSETS

File: contracts/libraries/Transitions.sol [hunk 2]
```diff
@@ -517,11 +517,11 @@ library Transitions {
...
             uint256 stakeAdjustmentFactor,
             uint64 startEpoch
-        ) = abi.decode((_rawBytes), (uint8, bytes32, uint32, uint32, uint32[], uint256[], uint256, uint64));
+        ) = abi.decode((_rawBytes), (uint8, bytes32, uint32, uint32, uint32[5], uint256[5], uint256, uint64));
```
Comment (xiaozhou) [inline contracts/libraries/Transitions.sol:524]: MAX_REWARD_ASSETS

File: contracts/libraries/Transitions.sol [hunk 3]
```diff
@@ -540,9 +540,9 @@ library Transitions {
         pure
         returns (DataTypes.UpdatePoolTransition memory)
     {
-        (uint8 transitionType, bytes32 stateRoot, uint32 poolId, uint256[] memory rewardPerEpoch) = abi.decode(
+        (uint8 transitionType, bytes32 stateRoot, uint32 poolId, uint256[5] memory rewardPerEpoch) = abi.decode(
```
Comment (xiaozhou) [inline contracts/libraries/Transitions.sol:543]: MAX_REWARD_ASSETS

File: contracts/libraries/Transitions.sol [hunk 4]
```diff
@@ -540,9 +540,9 @@ library Transitions {
...
             (_rawBytes),
-            (uint8, bytes32, uint32, uint256[])
+            (uint8, bytes32, uint32, uint256[5])
```
Comment (xiaozhou) [inline contracts/libraries/Transitions.sol:545]: MAX_REWARD_ASSETS

File: contracts/libraries/Transitions.sol (hunk 3 above)
Comment (Dominator008) [reply contracts/libraries/Transitions.sol]: Done.

File: contracts/libraries/Transitions.sol (hunk 4 above)
Comment (Dominator008) [reply contracts/libraries/Transitions.sol]: Done.

File: contracts/libraries/Transitions.sol (hunk 2 above)
Comment (Dominator008) [reply contracts/libraries/Transitions.sol]: Done.

File: contracts/libraries/Transitions.sol (hunk 1 above)
Comment (Dominator008) [reply contracts/libraries/Transitions.sol]: Done.

**Pull request merged**